nntrainer/dataset/databuffer_factory.cpp

```diff
@@ -23,14 +23,12 @@ namespace nntrainer {
  */
 std::unique_ptr<DataBuffer> createDataBuffer(DatasetType type) {
   switch (type) {
-  case DatasetType::GENERATOR:
-    return std::make_unique<DataBufferFromCallback>();
   case DatasetType::FILE:
```

Comment:
so, the default type is file?

Reply:
Generator is forced to have a callback argument, so it is not supported when only type is given

Reply:
got it.

Reply:
> Generator is forced to have a callback argument, so it is not supported when only type is given

It might be better to keep the same constraints for both cb/file datasets for consistency.

Reply:
Yeah, I think we could decide this later, I'd like to keep the structure for reference for now :)